e2e/basic/basic_test.go

```diff
@@ -1,53 +1,59 @@
 package basic
 
 import (
-	"testing"
+	"context"
 
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
 
-	"github.com/celestiaorg/knuu/pkg/knuu"
+	"github.com/celestiaorg/knuu/pkg/instance"
 )
 
-func TestBasic(t *testing.T) {
-	t.Parallel()
+const (
+	testImage = "alpine:latest"
+)
+
+func (ts *TestSuite) TestBasic() {
+	ts.T().Parallel()
 	// Setup
 
-	instance, err := knuu.NewInstance("alpine")
-	if err != nil {
-		t.Fatalf("Error creating instance '%v':", err)
-	}
-	err = instance.SetImage("docker.io/alpine:latest")
-	if err != nil {
-		t.Fatalf("Error setting image: %v", err)
-	}
-	err = instance.SetCommand("sleep", "infinity")
-	if err != nil {
-		t.Fatalf("Error setting command: %v", err)
-	}
-	err = instance.Commit()
-	if err != nil {
-		t.Fatalf("Error committing instance: %v", err)
-	}
+	ctx := context.Background()
 
-	t.Cleanup(func() {
-		require.NoError(t, knuu.BatchDestroy(instance))
+	target, err := ts.Knuu.NewInstance("alpine")
+	ts.Require().NoError(err)
+
+	ts.Require().NoError(target.SetImage(ctx, testImage))
+	ts.Require().NoError(target.SetCommand("sleep", "infinity"))
+	ts.Require().NoError(target.Commit())
+
+	ts.T().Cleanup(func() {
+		ts.T().Log("Tearing down Basic Test...")
+		err := instance.BatchDestroy(ctx, target)
+		if err != nil {
+			ts.T().Logf("error destroying instances: %v", err)
+		}
 	})
 
-	// Test logic
+	// Test Logic
+	ts.Require().NoError(target.Start(ctx))
+	ts.Require().NoError(target.WaitInstanceIsRunning(ctx))
 
-	err = instance.Start()
-	if err != nil {
-		t.Fatalf("Error starting instance: %v", err)
+	// Perform the test
+	type testCase struct {
+		name string
 	}
-	err = instance.WaitInstanceIsRunning()
-	if err != nil {
-		t.Fatalf("Error waiting for instance to be running: %v", err)
+
+	tt := []testCase{
+		{"Hello World"},
 	}
-	wget, err := instance.ExecuteCommand("echo", "Hello World!")
-	if err != nil {
-		t.Fatalf("Error executing command '%v':", err)
+
+	for _, tc := range tt {
+		tc := tc
+		ts.Run(tc.name, func() {
+			output, err := target.ExecuteCommand(ctx, "echo", tc.name)
+			ts.Require().NoError(err)
+
+			assert.Contains(ts.T(), output, tc.name)
+		})
```

Comment:
you don't need a for loop if you just have a single test case. just test it directly.
```suggestion
                input := "Hello World"
		ts.Run(tc.name, func() {
			output, err := target.ExecuteCommand(ctx, "echo", input)
			ts.Require().NoError(err)

			assert.Contains(ts.T(), output, input)
		})
```

Reply:
prep for having multiple tests